### PR TITLE
Add option to set a fixed dithering amount

### PR DESCRIPTION
**Pull request by zhouconghao**
This enables testing the pixel interpolation effect with a fixed offset between the SE WCS and coadd WCS. Let me know if I should come up with a test or change anything. Thanks!

### CODE REVIEW

**Comment by zhouconghao** [comment]
Somehow the test got an error in `conda install`. Pasted here

```
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
warning  libmamba Could not parse mod/etag header
warning  libmamba Could not parse mod/etag header
warning  libmamba Could not parse mod/etag header
warning  libmamba Could not parse mod/etag header
warning  libmamba Could not parse mod/etag header
warning  libmamba Could not parse mod/etag header
Download error (28) Timeout was reached [https://conda.anaconda.org/conda-forge/linux-64/htcondor-utils-23.0.3-h841b71b_0.conda]
Operation timed out after 9053 milliseconds with 0 bytes received
error    libmamba Error when extracting package: std::bad_alloc
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
/home/runner/work/_temp/677bb002-520b-44d9-8240-544f958b1106.sh: line 9:  1999 Aborted                 (core dumped) mamba install -q --file requirements.txt
Error: Process completed with exit code 134.
```

will try later.

File: descwl_shear_sims/wcs/sewcs.py [hunk 1]
```diff
@@ -41,12 +44,24 @@ def make_se_wcs(
...
-        dither_range = 0.5
-        off = rng.uniform(low=-dither_range, high=dither_range, size=2)
-        offset = galsim.PositionD(x=off[0], y=off[1])
+
+        if dither_amp is not None:
+            assert (
+                dither_amp > 0 and dither_amp < 1
+            ), "dither_amp should be in range (0, 1)"
+            sign = 1 if rng.uniform() > 0.5 else -1
+            offset = galsim.PositionD(x=sign * dither_amp, y=sign * dither_amp)
```
[the remark this answers is not in the document]
Comment (zhouconghao) [reply descwl_shear_sims/wcs/sewcs.py]: Right. Sorry got confused about the random seed. Fixed!

File: descwl_shear_sims/sim.py [hunk 2]
```diff
@@ -217,7 +223,6 @@ def make_sim(
             coadd_scale=coadd_scale,
             coadd_dim=coadd_dim,
             se_scale=pixel_scale,
-            dither=dither,
```
[the remark this answers is not in the document]
Comment (zhouconghao) [reply descwl_shear_sims/sim.py:220]: The argument was not used in the function. Happy to add it back!

File: descwl_shear_sims/wcs/sewcs.py [hunk 3]
```diff
@@ -41,12 +44,23 @@ def make_se_wcs(
...
+        if dither_size is not None:
+            assert (
+                dither_size > 0 and dither_size < 1
+            ), "dither_size should be in range (0, 1)"
+            offset = galsim.PositionD(x=dither_size, y=dither_size)
+        else:
+            dither_range = 0.5
+            off = rng.uniform(low=-dither_range, high=dither_range, size=2)
+            offset = galsim.PositionD(x=off[0], y=off[1])
+
```
Comment (beckermr) [inline descwl_shear_sims/wcs/sewcs.py:63]:
```suggestion

        if dither_size is None:
            dither_size = 0.5

        off = rng.uniform(low=-dither_size, high=dither_size, size=2)
        offset = galsim.PositionD(x=off[0], y=off[1])

```

We want this.

File: descwl_shear_sims/sim.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (beckermr) [reply descwl_shear_sims/sim.py:220]: fine to remove - just curious

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -25,6 +25,13 @@ jobs:
           channel-priority: strict
           show-channel-urls: true
 
+      - name: roll back libmamba?
+        shell: bash -l {0}
+        run:
```
Comment (beckermr) [inline .github/workflows/test.yml:30]:
```suggestion
        run: |
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -25,6 +25,13 @@ jobs:
           channel-priority: strict
           show-channel-urls: true
 
+      - name: roll back libmamba?
+        shell: bash -l {0}
+        run: |
+          conda activate base
+          conda list
+          conda install "libmamba<2" --yes
+
```
Comment (beckermr) [inline .github/workflows/test.yml:34]:
```suggestion
```

**Review by beckermr — APPROVED** [review]

File: descwl_shear_sims/sim.py [hunk 6]
```diff
@@ -106,6 +107,11 @@ def make_sim(
         Dimensions of psf image.  Default 51
     dither: bool, optional
         Whether to dither the images at the pixel level, default False
+    dither_size: float, optional
+        The amplitude of dithering in unit of a fraction of a pixel
+        for testing pixel interpolation.
+        All se WCS will be dithered by this amount in both +x and +y directions.
```
Comment (esheldon) [inline descwl_shear_sims/sim.py:113]: I think this should say "will be given random dithers with this amplitude in both..."

Comment (esheldon) [reply descwl_shear_sims/sim.py:113]: Also this actually defaults to 0.5.  I think this should also follow the same convention as other keywords and have the default in the function declaration.

File: descwl_shear_sims/wcs/sewcs.py [hunk 7]
```diff
@@ -41,12 +44,19 @@ def make_se_wcs(
...
+        raise ValueError("dither_size is set but dither is False")
+
     if dither:
         # do a small offset of the origin
         assert rng is not None
-        dither_range = 0.5
-        off = rng.uniform(low=-dither_range, high=dither_range, size=2)
+
+        if dither_size is None:
+            dither_size = 0.5
```
Comment (esheldon) [inline descwl_shear_sims/wcs/sewcs.py:55]: I thjink this default should be visible in the `make_sim` declaration

**Review by esheldon — CHANGES_REQUESTED** [review]

File: descwl_shear_sims/wcs/sewcs.py (hunk 7 above)
Comment (zhouconghao) [reply descwl_shear_sims/wcs/sewcs.py:55]: Fixed!

**Review by esheldon — APPROVED** [review]

**Pull request merged**